sdk/core/corehttp/tests/perf_tests/upload_binary.py

```diff
@@ -29,7 +29,14 @@ def __init__(self, arguments):
         blob_name = "uploadtest"
         self.blob_endpoint = f"{self.account_endpoint}{self.container_name}/{blob_name}"
         self.upload_stream = RandomStream(self.args.size)
-        self.upload_stream_async = AsyncRandomStream(self.args.size)
+
+        # The AsyncIteratorRandomStream is used for upload stream scenario, since the
+        # async httpx transport requires the request body stream to be type AsyncIterator (i.e. have an __aiter__ and __anext__ method rather than __iter__).
```

Comment:
Just so the comment reflects accurately, looks like it only needs the `__aiter`__ since the check is for `AsyncIterable`. Albeit, an AsyncIterator is also an AsyncIterable, but seems we technically don't need the object to have an `__anext__` method.

Reply:
Hmm, so I get this error when I try removing __anext__: `Error: Can't instantiate abstract class AsyncIteratorRandomStream with abstract method __anext__`

Not quite sure how I would remove that method here. Do you have any thoughts?

Reply:
updated thanks!